pallets/loan/src/lib.rs

```diff
@@ -434,6 +503,74 @@ pub mod pallet {
 			Self::deposit_event(Event::<T>::NAVUpdated(pool_id, updated_nav));
 			Ok(())
 		}
+
+		/// a call to add a new write off group for a given pool
+		/// write off groups are always append only
+		#[pallet::weight(100_000)]
+		#[transactional]
+		pub fn add_write_off_group(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			group: WriteOffGroup<T::Rate>,
+		) -> DispatchResult {
+			// ensure this is coming from an admin origin
+			<T as Config>::AdminOrigin::ensure_origin(origin)?;
+
+			// check if the pool exists
+			pallet_pool::Pallet::<T>::check_pool(pool_id)?;
+
+			// ensure write off percentage is not more than 100
+			ensure!(
+				group.percentage <= One::one(),
+				Error::<T>::ErrInvalidWriteOffGroup
```

Comment:
I would find it helpful to be more specific in the naming of this variant and make it `ErrInvalidWriteOffGroupPercentage`.

Reply:
We do not want to be too specific about it IMO